app/javascript/components/subject/CaseStatus.js

```diff
@@ -35,32 +37,59 @@ class CaseStatus extends React.Component {
       this.setState({ [event.target.id]: value }, () => {
         if (event.target.id === 'case_status') {
           const confirmedOrProbable = value === 'Confirmed' || value === 'Probable';
-          const hideModal = this.state.isolation && confirmedOrProbable && this.props.patient.case_status !== '';
-
-          this.setState({ showCaseStatusModal: !hideModal, confirmedOrProbable }, () => {
-            // specific case where case status is just changed with no modal
-            if (hideModal) {
-              this.submit();
-            }
-            if (!confirmedOrProbable && this.state.case_status !== '') {
-              this.setState({ isolation: false });
+          this.setState({ showCaseStatusModal: true, confirmedOrProbable }, () => {
+            if (!this.props.patient.monitoring) {
+              this.setState({
+                modal_text: `Are you sure you want to change case status from ${this.props.patient.case_status} to ${value ||
+                  'blank'}? Since this record is on the Closed line list, updating this value will not move this record to another line list. If this individual should be actively monitored, please update the record’s Monitoring Status.`,
+              });
+            } else if (value === '') {
+              this.setState({
+                modal_text: `Are you sure you want to change case status from ${this.props.patient.case_status} to blank? The monitoree will remain in the same workflow.`,
+              });
+            } else if (
+              this.state.isolation &&
+              !confirmedOrProbable &&
+              (this.props.patient.case_status === 'Confirmed' || this.props.patient.case_status === 'Probable')
+            ) {
+              this.setState({
```

Comment:
You should be setting `isolation` to false here as well. Make sure you update the test case for this.

 Currently if you go from a confirmed or probable case status in the isolation workflow and then select any of the other case status options (aside from blank), it will stay isolation even though this message is shown.

Reply:
done